imessage/bluebubbles/api.go

```diff
@@ -0,0 +1,1631 @@
+package bluebubbles
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+	"unicode"
+
+	"github.com/gorilla/websocket"
+	"github.com/rs/zerolog"
+	"github.com/sahilm/fuzzy"
+	"maunium.net/go/mautrix/id"
+
+	"go.mau.fi/mautrix-imessage/imessage"
+)
+
+const (
+	// Ref: https://github.com/BlueBubblesApp/bluebubbles-server/blob/master/packages/server/src/server/events.ts
+	NewMessage            string = "new-message"
+	MessageSendError      string = "message-send-error"
+	MessageUpdated        string = "updated-message"
+	ParticipantRemoved    string = "participant-removed"
+	ParticipantAdded      string = "participant-added"
+	ParticipantLeft       string = "participant-left"
+	GroupIconChanged      string = "group-icon-changed"
+	GroupIconRemoved      string = "group-icon-removed"
+	ChatReadStatusChanged string = "chat-read-status-changed"
+	TypingIndicator       string = "typing-indicator"
+	GroupNameChanged      string = "group-name-change"
+	IMessageAliasRemoved  string = "imessage-alias-removed"
+)
+
+type blueBubbles struct {
+	bridge            imessage.Bridge
+	log               zerolog.Logger
+	ws                *websocket.Conn
+	messageChan       chan *imessage.Message
+	receiptChan       chan *imessage.ReadReceipt
+	typingChan        chan *imessage.TypingNotification
+	chatChan          chan *imessage.ChatInfo
+	contactChan       chan *imessage.Contact
+	messageStatusChan chan *imessage.SendMessageStatus
+	backfillTaskChan  chan *imessage.BackfillTask
+
+	contactsLastRefresh time.Time
+	contacts            []Contact
+
+	bbRequestLock sync.Mutex
+
+	usingPrivateApi bool
```

Comment:
https://go.dev/wiki/CodeReviewComments#initialisms

Reply:
Have I gone too far with this commit?

https://github.com/mautrix/imessage/pull/189/commits/2c1a437bf33179bf55945f8649e03f9e73382987